### PR TITLE
Merge: Code fixes

### PR DESCRIPTION
**Pull request by Chemin1**
## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildservers are happy.
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: src/libs/merge/kdbmerge.c [hunk 1]
```diff
@@ -46,10 +46,11 @@ static int getStatisticalValue (Key * informationKey, char * metaName)
  * @param informationKey contains the statistics in its meta information
  * @param metaName which statistic to set
  * @param value which value to set it to, must be a number
+ * @returns 0 on success, 1 otherwise.
```
Comment (markus2330) [inline src/libs/merge/kdbmerge.c]: You can use retval here.

File: src/libs/merge/kdbmerge.c [hunk 2]
```diff
@@ -69,14 +70,15 @@ static void setStatisticalValue (Key * informationKey, char * metaName, int valu
 						     "Statistical value %d was too large for its buffer. This happened with meta name %s.",
 						     value, metaName);
 		}
-		return;
+		return 1;
```
Comment (markus2330) [inline src/libs/merge/kdbmerge.c]: Usually, we use -1 for errors.

**Comment by Chemin1** [comment]
jenkins build libelektra please

File: src/libs/merge/kdbmerge.c [hunk 3]
```diff
@@ -46,7 +46,7 @@ static int getStatisticalValue (Key * informationKey, char * metaName)
  * @param informationKey contains the statistics in its meta information
  * @param metaName which statistic to set
  * @param value which value to set it to, must be a number
- * @returns 0 on success, 1 otherwise.
+ * @retval 0 on success, -1 otherwise.
```
Comment (markus2330) [inline src/libs/merge/kdbmerge.c]: for every different retval you can make its own line.

Comment (Chemin1) [reply src/libs/merge/kdbmerge.c]: Is this also true for jna bindings? There it is in one line as well. If yes, we could make a good first issue out of it.

Comment (markus2330) [reply src/libs/merge/kdbmerge.c]: Thank you very much for the offer! But such a small thing is not worth the effort of making a "good first issue".

**Comment by markus2330** [comment]
Seems like conflicts with doc/news/_preparation_next_release.md are still possible. Can you debug this?

**Comment by Chemin1** [comment]
This must be a problem from GitHub. Merging works flawlessly on the command line

```zsh
➜  libelektra git:(master) ✗ git reset --hard HEAD
HEAD is now at bab98c8c1 Link Checker: Ignore Jenkins Wiki URLs
➜  libelektra git:(master) ✗ git checkout -b masterTest
Switched to a new branch 'masterTest'
➜  libelektra git:(masterTest) ✗ git rev-parse origin/smallMergeFixes
ab90f57612f495c1ca94e0c4f315cd40d20ffe9a
➜  libelektra git:(masterTest) ✗ git merge origin/smallMergeFixes
Auto-merging src/libs/merge/kdbmerge.c
Auto-merging doc/tutorials/cmerge.md
Auto-merging doc/news/_preparation_next_release.md
Merge made by the 'recursive' strategy.
 doc/news/_preparation_next_release.md |  1 +
 doc/tutorials/cmerge.md               |  4 ++--
 src/libs/merge/kdbmerge.c             | 16 ++++++++++------
 src/tools/kdb/cmerge.cpp              |  4 +---
 4 files changed, 14 insertions(+), 11 deletions(-)
```
and `git show` gives
```
commit f55a4fc8fa41d377eb71704f5b8b63443dc24fa3 (HEAD -> masterTest)
Merge: bab98c8c1 ab90f5761
Author: Dominic Jäger <dominic.jaeger@gmail.com>
Date:   Sun Nov 17 11:21:58 2019 +0100

    Merge remote-tracking branch 'origin/smallMergeFixes' into masterTest

diff --cc doc/news/_preparation_next_release.md
index b3e3e1f8a,f186a8af1..43be7baf1
--- a/doc/news/_preparation_next_release.md
+++ b/doc/news/_preparation_next_release.md
@@@ -342,7 -338,7 +342,8 @@@ you up to date with the multi-language 
  - Improved `range` plugin error message. _(Michael Zronek)_
  - Improved error codes documentation to clarify the hierarchy for developers. _(Michael Zronek)_
  - Release notes now use git's union merge driver. _(Dominic Jäger)_
 +- Please remove me. I'm only here for the build server. _(Dominic Jäger)_
+ - I'm only here for the build server. This PR contains only fix ups. _(Dominic Jäger)_
  
  ## Infrastructure
```

In contrast, when I remove the `.gitattributes` file on `masterTest` and commit the change then the following `git merge origin/smallMergeFixes` says `Automatic merge failed;`.

As I couldn't find a solution on their [help page](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-on-github), I followed GitHub's advice and sent a request to their support.

**Comment by markus2330** [comment]
Thank you! Maybe they do not implement ".gitattributes" yet?

Should I merge this now or let us leave it open so that the GitHub support can look at it?

**Comment by Chemin1** [comment]
> Should I merge this now or let us leave it open so that the GitHub support can look at it?

They have the `.gitattributes` file and diffs of the successful local merge and the conflict in the web UI => I assume merging this PR is OK.

**Comment by markus2330** [comment]
I rebased and merged the PR.

**Pull request closed without merging**